shell-ui/src/theme.js

```diff
@@ -0,0 +1,60 @@
+//@flow
+import React, { useContext, useState, type Node, useLayoutEffect } from 'react';
+import { defaultTheme } from '@scality/core-ui/dist/style/theme';
+import { THEME_CHANGED_EVENT } from '.';
+
+type ThemeName = 'dark' | 'light' | 'custom';
+type ThemeContextValues = {
+  themeName: ThemeName,
+  theme: Theme,
+  setTheme: (themeName: ThemeName) => void,
+  unSelectedThemes: ThemeName[],
+};
+const ThemeContext = React.createContext<ThemeContextValues | null>(null);
+
+const themes = ['dark', 'light'];
+
+type Theme = { brand: typeof defaultTheme.dark };
+
+export function useThemeName(): ThemeContextValues {
+  const themeContext = useContext(ThemeContext);
+  if (themeContext === null) {
+    throw new Error("useTheme hook can't be use outside <ThemeProvider/>");
+  }
+  return { ...themeContext };
+}
+
+export function ThemeProvider({
+  children,
+  onThemeChanged,
+}: {
+  children: (theme: Theme, themeName: ThemeName) => Node,
+  onThemeChanged?: (evt: CustomEvent) => void,
+}): Node {
+  const [themeName, setTheme] = useState<ThemeName>(
+    (localStorage.getItem('theme'): any) || 'dark',
+  );
+
+  const theme: Theme = {
+    brand: defaultTheme[themeName],
```

Comment:
Oh. So the theme isn't editable anymore, actually. So, for my previous comments, the "logo should be part of a theme" idea doesn't really apply. I'm a bit surprised we decided to remove this feature, but I guess it's not needed in short term.